frontend/src/lib/utils.tsx

```diff
@@ -110,6 +110,11 @@ export function deleteWithUndo({ undo = false, ...props }: Record<string, any>):
             onClick: () => {
                 deleteWithUndo({ undo: true, ...props })
             },
+            onClose: () => {
+                if (props.endpoint === 'annotation') {
+                    api.delete(`api/${props.endpoint}/${props.object.id}`)
+                }
```

Comment:
Personally a bit reluctant to put such special handling in general utils. I think it should be safe for all endpoints (there's just two other ones besides `annotation`: `insight` and `feature_flag`).